k8s-deployer/src/locks/http-client.ts

```diff
@@ -0,0 +1,91 @@
+import fetch, { RequestInit, Response } from "node-fetch"
+
+import { logger } from "../logger.js"
+
+export type FetchParams = {
+  endpoint: string,
+  options: RequestInit
+}
+
+export type RetryOptions = {
+  retries: number,
+  retryDelay: number,
+  fetchParams: FetchParams
+}
+
+export class HttpErrorOptions {
+  cause?: Error
+  responseData?: string | unknown
+}
+
+export class HttpError extends Error {
+  readonly type: string = "HttpError"
+  responseData?: string | unknown
+  constructor(
+    readonly method: string,
+    readonly endpoint: string,
+    readonly status: number,
+    readonly text: string,
+    private options: HttpErrorOptions
+  ){
+    super(text, { cause: options.cause })
+    this.responseData = options.responseData
+  }
+
+  toString(): string {
+    return JSON.stringify({ method: this.method, endpoint: this.endpoint, status: this.status, text: this.text, responseData: this.responseData, cause: this.cause })
+  }
+}
+
+export const invoke = async (options: RetryOptions, apiBody: Object) => {
+  let attempts = 1
+  while (options.retries > 0) {
+    logger.info("http-client.invoke(): attempt %s of %s invoking %s", attempts, options.retries, JSON.stringify(options.fetchParams))
+    try {
+        return await invokeApi(options.fetchParams, apiBody)
+    } catch (error: unknown) {
+      logger.warn("http-client.invoke(): attempt %s of %s invoking %s. Error: %s", attempts, options.retries, JSON.stringify(options.fetchParams), error)
+      if (attempts < options.retries) {
+        attempts++
+        await sleep(options.retryDelay)
```

Comment:
Because we have this sleep in the `invoke` method, maybe we should remove this sleep [while trying to acquire lock](https://github.com/kindredgroup/pit-toolkit/blob/447a48a541a81231e15791135b95b2dc9a2bea17/k8s-deployer/src/locks/lock-manager.ts#L73),  else we are going to sleep longer duration than necessary.

Reply:
Yes, makes sense to remove the second one.